### DISCUSSION TITLE
Error in language identification using whisper large

### DISCUSSION THREAD

**Discussion by Aafiya-H** [openai/whisper#2040] [Q&A]
Hello,
While performing language identification on Fleurs, I receive the following error message:

```
RuntimeError                              Traceback (most recent call last)
[<ipython-input-10-3c79e22d458b>](https://localhost:8080/#) in <cell line: 6>()
     14     audio = whisper.pad_or_trim(audio)
     15     mel = whisper.log_mel_spectrogram(audio).to(model_large.device)
---> 16     _, probs = model_large.detect_language(mel)
     17     detected_language.append(max(probs, key=probs.get))
     18     # print(f"Detected language: {max(probs, key=probs.get)}")

9 frames
[/usr/local/lib/python3.10/dist-packages/torch/nn/modules/conv.py](https://localhost:8080/#) in _conv_forward(self, input, weight, bias)
    304                             weight, bias, self.stride,
    305                             _single(0), self.dilation, self.groups)
--> 306         return F.conv1d(input, weight, bias, self.stride,
    307                         self.padding, self.dilation, self.groups)
    308 

RuntimeError: Given groups=1, weight of size [1280, 128, 3], expected input[1, 80, 3000] to have 128 channels, but got 80 channels instead

```
The code snippet works for other sizes of Whisper. Could you please guide me on how to fix this?

**Comment by glangford** [comment]
From the announcement of large-v3,
> "The input uses 128 Mel frequency bins instead of 80" ...
https://github.com/openai/whisper/discussions/1762

By default calling `log_mel_spectrogram` directly without extra parameters will use 80 and from the error message I suspect this is the problem. You can override this when calling the low level API, here is the function signature:

https://github.com/openai/whisper/blob/ba3f3cd54b0e5b8ce1ab3de13e32122d0d5f98ab/whisper/audio.py#L110
